app/global/MainModule.java

```diff
@@ -393,6 +396,18 @@ private Clock provideClock() {
         return Clock.systemUTC();
     }
 
+    @Provides
+    @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD") // Invoked reflectively by Guice
+    private JobRefSerializer provideJobRefSerializer() {
+        // This must match the configuration of each JobCoordinator, or else
+        // we will not be able to guarantee that job actors will run after
+        // a shard moves within the cluster.
+        return new TwoWayJobRefSerializer(
+                ImmutableList.of(AlertJobRepository.class, ReportRepository.class),
```

Comment:
So this is good except `AlertJobRepository` is a repository wrapper. The serializability of the job reference / repository is determined by the repository actually wrapped by it. 

-- also, should it not come from configuration?

In general, I get where you're going with this and it's good, but I think it might be more than we want to bite off given the class structure of the repositories. Defense against bad configuration is good, but if you're going to setup clustered reporting/alerting and now have a remote repository you're in deep trouble already.

Reply:
Mentioned above, I'm just stripping this out for now since I agree it's not worth the ROI due to the complexity from clustering.